src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -458,7 +458,8 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     if (
       current === null ||
       current.tag !== HostPortal ||
-      current.stateNode.containerInfo !== portal.containerInfo ||
+      // Persistent containers don't have permanent identity.
+      // current.stateNode.containerInfo !== portal.containerInfo ||
```

Comment:
@gaearon This is a breaking change. If you create a portal with a different container in the same slot, with the same key, you'll probably start seeing errors in mutable mode.

I'm not really sure what's best here.

Reply:
Why did you do it then? 😛

Reply:
In this design, persistent containers don't have permanent identity. They get replaced all the time. Practically that means that probably means the container most often has a reference to its own "atom" like we do with HostRoot fibers.

So I don't actually really know what you would pass in to createPortal in this world. I guess any container would work as long as it has the same internal slot that represents the same atom pointer.

Maybe this whole thing is a poor design and really you need container to be a permanent identity and a new concept that gets swapped out.